docs/spec/sources.md

```diff
@@ -362,5 +376,39 @@ _Structured Content Mode_ of the HTTP Protocol Binding for CloudEvents. Sources
 SHOULD send events to its
 [Destination](https://pkg.go.dev/github.com/knative/pkg/apis/duck/v1?tab=doc#Destination).
 
+### CloudEvent `knsource` extension
+
+Sources SHOULD produce CloudEvents with the `knsource` CloudEvent extension
+identifying the originating [Source Custom Object](#source-custom-objects).
+
+The `knsource` value SHOULD be of the form `<name>.<plural>.<group>`, where:
```

Comment:
```suggestion
The `knsource` value MUST be of the form `<name>.<plural>.<group>`, where:
```

If there is no known way to understand this value, it likely has no meaning. 

I would recommend inventing a uri for this instead of a magic string, like `k8s://<plural>.<group>/<namespace>/<name>/<uid>` or something like that.

But l think likely if you care this much about this level of detail of _who_ you might also care about the pod that is doing the work.

Reply:
The reason it is SHOULD and not MUST is the value can be overridden. What about `The default knsource value MUST ...`?

We can invent a magic uri if you want, it does not make a difference IMO. Independently of the actual format, I think we agree that `plural`, `group` and `name` should be in there. 

`uid` is a dynamic value making the deployment of Triggers much more complicated. It's also much easier for the user to understand where the events are coming from if the value only contains elements (name, plurals, group) that are specified. Maybe `<plural>` should even be `<kind>` (lower-case).

As for `namespace`, it's a bit more complicated. I'd like to be able to specify filters that are namespace-unaware. I can't think of a better solution than to add another extension, `knsourcens`, at least until we have a better filtering language.